src/aqua/chain/blocks.aqua

```diff
@@ -0,0 +1,58 @@
+module Blocks declares get_latest, lte, gt
+
+import get_string, spell_log from "../fluence/spell.aqua"
+import AuroraInfo from "../chain/aurora.aqua"
+import SpellId from "../types.aqua"
+
+import FluenceAuroraConnector from "services.aqua"
+
+-- Get latest block number from RPC
+func get_latest(spell_id: SpellId, chain: AuroraInfo) -> ?string:
+    latest: ?string
+    r <- FluenceAuroraConnector.latest_block_number(chain.api_endpoint)
+    if r.success:
+        latest <<- r.result
+    else:
+        spell_log(spell_id, ["error retrieving latest block number", r.error])
+    <- latest
+
+-- implements Less Than or Equal for two hex strings, i.e. `hex_a <= hex_b`
+-- returns nil if either `hex_a` or `hex_b` is an invalid hex
+func lte(spell_id: string, hex_a: string, hex_b: string) -> ?bool:
+    lte: ?bool
+
+    cmp <- FluenceAuroraConnector.hex_cmp(hex_a, hex_b)
+    if cmp.success:
+        -- `hex_a` is less than `hex_b`
+        if cmp.ordering == -1:
+            lte <<- true
+        -- `hex_a` equals to `hex_b`
+        if cmp.ordering == 0:
+            lte <<- true
+        -- `hex_a` is greater than `hex_b`
+        if cmp.ordering == 1:
+            lte <<- false
+    else:
+        spell_log(spell_id, ["hex_cmp error", cmp.error])
+
+    <- lte
+
+-- `hex_a > hex_b`
+func gt(spell_id: string, hex_a: string, hex_b: string) -> ?bool:
+    gt: ?bool
+
+    cmp <- FluenceAuroraConnector.hex_cmp(hex_a, hex_b)
+    if cmp.success:
+        -- `hex_a` is less than `hex_b`
+        if cmp.ordering == -1:
+            gt <<- false
+        -- `hex_a` equals to `hex_b`
+        if cmp.ordering == 0:
+            gt <<- false
+        -- `hex_a` is greater than `hex_b`
+        if cmp.ordering == 1:
+            gt <<- true
+    else:
+        spell_log(spell_id, ["hex_cmp error", cmp.error])
```

Comment:
Would nice to express via lte, mb?

Reply:
I tried, but it doesn't seem very optimal. Too many FFI calls, in my opinion.